File: readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    // Prompt the user for a string of text
    string text = get_string("Text: ");

    // Number of letters in text
    int letterCount = count_letters(text);

    // Number of words in text
    int wordCount = count_words(text);

    // Number of sentences in text
    int sentenceCount = count_sentences(text);

    // Plug values into Coleman-Liau index
    float L = ((float) letterCount / wordCount) * 100;
    float S = ((float) sentenceCount / wordCount) * 100;
    float index_float = (float) 0.0588 * L - 0.296 * S - 15.8;
    int index = round(index_float);

    // Print grade level
    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

// Function to count letters
int count_letters(string text)
{
    int count = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (isalnum(text[i]))
        {
            count++;
        }
    }
    return count;
}

// Function to count words
int count_words(string text)
{
    int count = 1;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (text[i] == ' ')
        {
            count ++;
        }
    }
    return count;
}

// Function to count sentences
int count_sentences(string text)
{
    int count = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            count++;
        }
    }
    return count;
}

File: sentimental-readability/readability.py
from cs50 import get_string


# Function to count letters
def countLetters(text):
    count = 0
    textLength = len(text)
    for i in range(textLength):
        if text[i].isalpha():
            count += 1
    return count


# Function to count words
def countWords(text):
    count = 1
    textLength = len(text)
    for i in range(textLength):
        if text[i] == " ":
            count += 1
    return count


# Function to count sentences
def countSentences(text):
    count = 0
    textLength = len(text)
    for i in range(textLength):
        if text[i] == "." or text[i] == "!" or text[i] == "?":
            count += 1
    return count


def main():
    # Prompt the user for a string of text
    text = get_string("Text: ")

    # Number of letters in text
    letterCount = countLetters(text)

    # Number of words in text
    wordCount = countWords(text)

    # Number of sentences in text
    sentenceCount = countSentences(text)

    # Plug values into Coleman-Liau index
    L = (letterCount / wordCount) * 100
    S = (sentenceCount / wordCount) * 100
    indexFloat = 0.0588 * L - 0.296 * S - 15.8
    index = round(indexFloat)

    # Print grade level
    if index >= 16:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {index}")


main()
